churaverse-plugins-server/src/churarenPlugin/churarenAlchemyItems/trapPlugin/controller/socketController.ts

```diff
@@ -0,0 +1,41 @@
+import { IMainScene, IEventBus, Store, Position, EntitySpawnEvent } from 'churaverse-engine-server'
+import { BaseSocketController } from '@churaverse/network-plugin-server/interface/baseSocketController'
+import { RegisterMessageListenerEvent } from '@churaverse/network-plugin-server/event/registerMessageListenerEvent'
+import { RegisterMessageEvent } from '@churaverse/network-plugin-server/event/registerMessageEvent'
+import { IMessageListenerRegister } from '@churaverse/network-plugin-server/interface/IMessageListenerRegister'
+import { TrapSpawnMessage } from '../message/trapSpawnMessage'
+import { Trap } from '../domain/trap'
+import { TrapHitMessage } from '../message/trapHitMessage'
+
+export class SocketController extends BaseSocketController<IMainScene> {
+  private messageListenerRegister!: IMessageListenerRegister<IMainScene>
+
+  public constructor(eventBus: IEventBus<IMainScene>, store: Store<IMainScene>) {
+    super(eventBus, store)
+  }
+
+  public registerMessage(ev: RegisterMessageEvent<IMainScene>): void {
+    ev.messageRegister.registerMessage('trapSpawn', TrapSpawnMessage, 'others')
+    ev.messageRegister.registerMessage('trapHit', TrapHitMessage, 'allClients')
+  }
+
+  public setupRegisterMessageListener(ev: RegisterMessageListenerEvent<IMainScene>): void {
+    this.messageListenerRegister = ev.messageListenerRegister
+  }
+
+  public registerMessageListener(): void {
+    this.messageListenerRegister.on('trapSpawn', this.trapSpawn.bind(this))
+  }
+
+  public unregisterMessageListener(): void {
+    this.messageListenerRegister.off('trapSpawn', this.trapSpawn.bind(this))
+  }
+
+  private trapSpawn(msg: TrapSpawnMessage, senderId: string): void {
```

Comment:
こちらバインドすると関数が複数実行されちゃうので、アロー関数で書いた方がいいですね

```suggestion
  private trapSpawn = (msg: TrapSpawnMessage, senderId: string): void => {
```